rosetta/create-t5x-distribution.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+set -euox pipefail
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+if [[ $# -ne 1 ]]; then
+  echo "[ERROR]: Need to pass a patchlist file: ./$0 <patchlist.txt>"
+  exit 1
+fi
+
+PATCH_LIST=$(readlink -f $1)
+
+# This is the git directory of t5x we will build the distribution in
+INSTALLED_T5X_DIR=${INSTALLED_T5X_DIR:-}
+# If provided, this is the commit that we'll base the distribution on
+DISTRIBUTION_BASE_REF=${DISTRIBUTION_BASE_REF:-}
+INTERNAL_T5X_SUBMODULE_DIR=$SCRIPT_DIR/t5x
+UPSTREAM_GIT_URL=https://github.com/google-research/t5x.git
+
+if [[ -z "$INSTALLED_T5X_DIR" ]]; then
+  echo "[ERROR]: Need to specify the env INSTALLED_T5X_DIR="
+  exit 1
+fi
+
+cd $INSTALLED_T5X_DIR
+git config user.email "jax@nvidia.com"
+git config user.name "NVIDIA"
+
+if [[ -z "$DISTRIBUTION_BASE_REF" ]]; then
+  # Note: not origin/main
+  DISTRIBUTION_BASE_REF=$(git rev-parse main)
+fi
+
+echo "[INFO]: Basing distribution on t5x commit: $DISTRIBUTION_BASE_REF"
+# Switch to main so other branches can be forced updated
+git switch main
+# Create a local branch to mark the base commit
+git branch --force distribution-base $DISTRIBUTION_BASE_REF
+# Create a local branch for the distribution that starts from the base
+git branch --force rosetta-distribution distribution-base
+git switch rosetta-distribution
+
+##############################
+# Update NV Internal patches #
+##############################
+# Create local branches for all remote branches since remotes aren't recognized
+# when adding local repos as remotes. Excludes origin/HEAD and origin/test_\d+
+function git-nv-t5x {
+  git -C $INTERNAL_T5X_SUBMODULE_DIR "$@"
+}
+NV_REMOTE_NAME=nv-t5x
+if [[ -d $INTERNAL_T5X_SUBMODULE_DIR ]]; then
+  for remote_branch in $(git-nv-t5x branch --remotes | grep -v 'origin/HEAD' | egrep -v 'origin/test_[0-9]+'); do
+    # Try creating a local tracking branch, but if already exists, then update it to match remote
+    git-nv-t5x checkout --track $remote_branch || git-nv-t5x reset --hard $remote_branch
```

Comment:
Why does the local branch need to track remote?

Reply:
I did this b/c when I `git remote add -f $NV_REMOTE_NAME  $INTERNAL_T5X_SUBMODULE_DIR`, only the local branches appear when you list branches from that remote (which happened to be a directory as opposed to a url). So the workaround was to pull the branches first so we have local tracking branches of all the remote branches.